src/haddock/gear/known_cns_errors.py

```diff
@@ -0,0 +1,158 @@
+"""Detect known/common CNS errors.
+
+Inspired from:
+https://github.com/haddocking/haddock25/blob/main/tools/check-error-messages.sh
+"""
+
+from pathlib import Path
+
+from haddock.core.exceptions import KnownCNSError
+from haddock.core.typing import FilePath, Optional, Union
+
+# Dictionary of known errors
+# as key:    How to catch it in the cns.out
+# as value:  Message to user
+KNOWN_ERRORS = {
+    "CHAIN LENGTH FOR SYMMETRY RESTRAINTS DOES NOT MATCH": (
+        "Mismatch between chain length for symmetry restraints. "
+        "Check your input molecules and symmetry restraints."
+        ),
+    "NCS-restraints error encountered: Improperly defined non-crystallographic symmetry": (  # noqa : E501
+        "Improperly defined non-crystallographic symmetry (NCS). "
+        "Check your symmetry restraints definition."
+        ),
+    "error in SYMMETRY potential, check NOE table": (
+        "Check your symmetry restraints definition."
+        ),
+    "exceeded allocation for NOE-restraints": (
+        "Too many distance restraints defined. "
+        "Try to reduce this number by checking your definition of active "
+        "and passive residues. "
+        "Make sure to filter those for solvent accessibility."
```

Comment:
Add another suggestion:
"Or alternatively increase the nres parameter in the noe statements in the relevant CNS scripts."